flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/nfa/compiler/NFACompilerTest.java

```diff
@@ -42,6 +45,43 @@
 
 public class NFACompilerTest extends TestLogger {
 
+	@Rule
+	public ExpectedException expectedException = ExpectedException.none();
+
+	@Test
+	public void testNFACompilerUniquePatternName() {
+
+		// adjust the rule
+		expectedException.expect(MalformedPatternException.class);
+		expectedException.expectMessage("Duplicate pattern name: start. Pattern names must be unique.");
```

Comment:
I match on both type and message. Do you suggest to remove the message matching part or you have a better way to do so?